code/modules/mob/living/simple_animal/bot/mulebot.dm

```diff
@@ -829,6 +828,16 @@
 	if(.)
 		visible_message("<span class='notice'>[src]'s safeties are locked on.</span>")
 
+/mob/living/simple_animal/bot/mulebot/bot_step(dest)
```

Comment:
Does this still drain power if the bot is under a player's control?

Reply:
> 
> 
> Does this still drain power if the bot is under a player's control? I'd assume not

Oh yeah. Now I have to dig in and find a proc or signal called when both players and the bot tries to move, but not called when it is pushed or force moved by something like scrungulo_act

Reply:
I do not believe this is a trivial fix and is probably why it was in Moved() before.

I don't think there's a way to tell the difference between being moved because you're AI patrolling, being moved because the client sent a keypress, being moved because because you're being pushed or being moved because it's forced that'll catch all the edge cases. It's not even like a vehicle where the pAI is piloting it - the pAI becomes the mulebot.

Reply:
Issue resolves with signals.

Base bot movement code has been tweaked because I didn't realise mulebots use their own snowflake movement code. That has been tested and works too. And should be an improvement over the old code. Added special signals that bots send before moving and after moving. These same signals are also sent by the mulebot's snowflake movement code too, which drains the mulebot's battery and stops its movement if it should not be able to move.